### PR TITLE
FontAwesome 5.10.1

### PR DESCRIPTION
**Pull request by bgrochal**
Things done:
 - FontAwesome 5.10.1 integration,
 - sbt `copyAssets` task configured to copy `webfonts` directory from a FontAwesome WebJar to appropriate `assets` subdirectory (statically provided font files are no longer required, it is sufficient to bump `FontAwesome` dependency version now),
 - legacy `guide/project` directory deleted.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -443,6 +443,19 @@ lazy val `guide-guide` =
     Dependencies.guideJsDeps,
     Some((`guide-backend`, "io.udash.web.styles.GuideCssRenderer")),
     Def.task(Some((`guide-commons` / sourceDirectory).value / "main" / "assets"))
+  ).settings(
+    Compile / copyAssets := {
```
Comment (ddworak) [inline build.sbt]: I'd rather you add this logic to the existing `copyAssets` logic in `frontendExecutable` - FontAwesome is available on both frontends anyway via `guide-commons`.

File: build.sbt [hunk 2]
```diff
@@ -443,6 +443,19 @@ lazy val `guide-guide` =
...
+      // (Compile / copyAssets) logic already defined within `frontendExecutable` should be executed first
+      (Compile / copyAssets).value
+      // font-awesome WebJar is required on the classpath to execute the next step
+      (Assets / WebKeys.webJars).value
+      // copying font-awesome webfonts to a location required by font-awesome styles 
+      // (appropriate CSS file is inlined by less within UdashStatics/WebContent/guide/assets/styles/assets.min.css)
+      IO.copyDirectory(
+        (Assets / WebKeys.webJarsDirectory).value / "lib/font-awesome/webfonts",
```
Comment (ddworak) [inline build.sbt]: I believe `lib` here comes from `webModulesLib` (could change)

File: build.sbt [hunk 3]
```diff
@@ -443,6 +443,19 @@ lazy val `guide-guide` =
...
+        target.value / "UdashStatics/WebContent/guide/assets/webfonts"
```
Comment (ddworak) [inline build.sbt]: You could ensure this is the same `assets` dir as in other definitions (will be much easier once it's integrated into regular `copyAssets`)

File: bootstrap4/.js/src/main/scala/io/udash/bootstrap/utils/UdashIcons.scala [hunk 4]
```diff
@@ -1404,6 +1598,7 @@ object UdashIcons {
       def wrench: CssStyle = fas("wrench")
       def xRay: CssStyle = fas("x-ray")
       def yenSign: CssStyle = fas("yen-sign")
+      def yinYang: CssStyle = fas("yin-yang")
     }
```
Comment (ddworak) [inline bootstrap4/.js/src/main/scala/io/udash/bootstrap/utils/UdashIcons.scala]: Do you think we could add a test for rendering these based on e.g. second example `https://developer.mozilla.org/en-US/docs/Web/API/FontFaceSet/check#Examples`?

File: bootstrap4/.js/src/test/scala/io/udash/bootstrap/utils/UdashIconsTest.scala [hunk 5]
```diff
@@ -1,15 +1,21 @@
...
+import io.udash.testing.AsyncUdashCoreFrontendTest
+import org.scalajs.dom._
+import org.scalatest.Succeeded
+import scalatags.JsDom.all._
+
+import scala.concurrent.Future
+
+class UdashIconsTest extends AsyncUdashCoreFrontendTest with CssView {
   "Bootstrap icons" should {
+    document.body.appendChild(UdashBootstrap.loadFontAwesome())
```
Comment (ddworak) [inline bootstrap4/.js/src/test/scala/io/udash/bootstrap/utils/UdashIconsTest.scala]: I'd rather this was in `beforeAll`, no guarantees about when the constructor is called here at all

File: bootstrap4/.js/src/main/scala/io/udash/bootstrap/utils/UdashIcons.scala (hunk 4 above)
Comment (bgrochal) [reply bootstrap4/.js/src/main/scala/io/udash/bootstrap/utils/UdashIcons.scala]: According to the docs, the two-arg variant of this method

> does not check for individual glyph coverage

([explanation](http://lists.w3.org/Archives/Public/www-style/2015Aug/0330.html)). However I've finally managed to implement a test for FontAwesome API/statics compatibility, please check it out under `UdashIconsTest`.

**Pull request merged**